component/src/main/java/org/wso2/extension/siddhi/gpl/execution/streamingml/classification/hoeffdingtree/HoeffdingClassifierStreamProcessorExtension.java

```diff
@@ -0,0 +1,222 @@
+import org.wso2.siddhi.query.api.definition.AbstractDefinition;
+import org.wso2.siddhi.query.api.definition.Attribute;
+import org.wso2.siddhi.query.api.exception.SiddhiAppValidationException;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * Predict using a Hoeffding Adaptive Tree model.
+ * built via @{@link HoeffdingClassifierStreamProcessorExtension}
+ */
+@Extension(
+        name = "hoeffdingTreeClassifier",
+        namespace = "streamingml",
+        description = "Performs classification with Hoeffding Adaptive Tree for evolving "
+                + "data streams that uses ADWIN to replace branches for new ones.",
+        parameters = {
+                @Parameter(name = "model.name",
+                        description = "The name of the model to be used for prediction.",
+                        type = {DataType.STRING})
+        },
+        returnAttributes = {
+                @ReturnAttribute(name = "prediction",
+                        description = "Predicted class label",
+                        type = {DataType.STRING}),
+                @ReturnAttribute(name = "confidenceLevel",
+                        description = "Probability of the prediction",
+                        type = {DataType.DOUBLE})
+        },
+        examples = {
+                @Example(
+                        syntax = "define stream StreamA (attribute_0 double, attribute_1 double, "
+                                + "attribute_2 double, attribute_3 double);\n" +
+                                "\n"
+                                + "from StreamA#streamingml:hoeffdingTreeClassifier('model1', "
+                                + " attribute_0, attribute_1, attribute_2, attribute_3) \n"
+                                + "select attribute_0, attribute_1, attribute_2, attribute_3, "
+                                + "prediction, predictionConfidence insert into outputStream;",
+                        description = "A Hoeffding Tree model with the name 'model1' will be used "
+                                + "to predict the label of the feature vector represented "
+                                + "by attribute_0, attribute_1, attribute_2, attribute_3. "
+                                + "Predicted label (String/Bool) along with the Prediction Confidence "
+                                + "and the feature vector will be emitted to the outputStream. "
+                                + "The outputStream will have following definition; "
+                                + "(attribute_0 double, attribute_1 double, attribute_2"
+                                + " double, attribute_3 double, prediction string, "
+                                + "confidenceLevel double)."
+                )
+        }
+)
+public class HoeffdingClassifierStreamProcessorExtension extends StreamProcessor {
+    private static final int minNoOfFeatures = 2;
```

Comment:
Constant should be all caps with underscores as the word separator

Reply:
Fixed in [#24fd7b0cd55c5a5589ddf0093a3c3fbe277252f9](https://github.com/wso2-extensions/siddhi-gpl-execution-streamingml/pull/16/commits/24fd7b0cd55c5a5589ddf0093a3c3fbe277252f9)